src/components/AddContact/AddContact.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Location } from 'history'
+import { SnackbarProvider } from 'notistack'
+import { MemoryRouter, Route } from 'react-router'
+
+import { createContactHandler } from '../../lib/mutations/createContact.handlers'
+import { getRealmSelectableHandler } from '../../lib/queries/getRealmSelectable/getRealmSelectable.handlers'
+import { mswServer } from '../../mocks/mswServer'
+
+import { AddContact } from '.'
+
+describe('AddContact', () => {
+  it('creates a contact', async () => {
+    mswServer.use(getRealmSelectableHandler(), createContactHandler())
+    let testLocation: Location | undefined
+    const client = new QueryClient()
+    render(
```

Comment:
make use of `renderWithProviders` method so you don't need to provide the query client or the snack bar provider.